pkg/cmd/server/api/types.go

```diff
@@ -870,6 +871,33 @@ type AssetExtensionsConfig struct {
 	HTML5Mode bool
 }
 
+const (
+	SecretBlockType = "ENCRYPTED STRING"
+	KeyBlockType    = "ENCRYPTING KEY"
+)
+
+// ExternalizableString allows specifying a string inline, or externally via env var or file.
+// When it contains only a string value, it marshals to a simple JSON string.
+type ExternalizableString struct {
+	// ExternalizableStringSpec specifies the string value, or external location
+	ExternalizableStringSpec
+}
+
+// ExternalizableStringSpec specifies a string value, or external location
+type ExternalizableStringSpec struct {
```

Comment:
all file references (throughout, not just here) would either have to be proxied to configmap/secret references/keys somehow, or deferred and resolved at runtime. the env bit would have to be deferred to runtime as well.